### ISSUE TITLE
关于structural shift的一点疑惑

### ISSUE THREAD

**Issue by gaoweijun5** [PoorOtterBob/KDD126#1]
复现您代码的时候注意到SD数据集上训练集的的大小为605个节点，但是这个地方原文里面说的是’we select a certain number of nodes for training, which amounts to 550 in the SD dataset and 1809 in the GBA dataset‘，说的是训练的时候用了550个节点。同时当新节点比例增加时，550的数目也不是固定的，是原文表述问题吗。
此外这个地方的train和train_node有什么区别吗。求教，谢谢
![ZF0016_idPNptOMxrsu6WYABBIile7](https://github.com/user-attachments/assets/bd57080b-dcad-41e6-9aa2-ab3413f07dac)

**Comment by PoorOtterBob** [comment]
Weijun Gao 您好：十分感谢您对我们工作的关注！

以SD数据集举例：我们选择了550个节点进行固定节点，这550个节点在训练集、验证集和测试集中是保持不变的。同时，我们额外有三组55个节点分别分配给了训练集、验证集和测试集来模拟新增节点，即路网结构的改变。即三个集合中都有605个节点，但是其中55个节点是不同的。
由于我们对于节点的选取是随机的，因此train和train_node 分别代表着训练集中选择的节点在整体节点（716个节点）中的index，以及根据index选取后的具体节点。

欢迎任何疑问，亦可加联系方式详细交流 ~

The Q&A in English version:
Q: When reproducing your code, I noticed that the size of the training set on the SD dataset is 605 nodes, but the original article in this place says 'we select a certain number of nodes for training, which amounts to 550 in the SD dataset and 1809 in the GBA dataset', which means that 550 nodes are used for training. Also the number of 550 is not fixed when the proportion of new nodes increases, is it a problem with the original formulation. Also is there any difference between train and train_node in this place. Thank you for your help.

A: The SD dataset serves as a case in point. In this instance, 550 nodes were selected for fixed nodes, which were maintained at a constant level across the training set, validation set, and test set. Concurrently, three supplementary sets of 55 nodes have been designated for incorporation into the training set, validation set, and test set, thereby simulating the introduction of new nodes and alterations to the road network structure. In other words, the three sets collectively comprise 605 nodes, of which 55 are distinct. Given that our selection of nodes is randomized, the "train" and "train_node" variables represent the index of the nodes selected in the training set among the overall nodes (716 nodes) and the specific nodes after selection based on the index, respectively.

**Comment by PoorOtterBob** [comment]
【补充说明】
Weijun Gao 您好：十分感谢您对我们工作的关注！
我想要补充一下，以SD数据集为例，550个节点当中只有495个节点是贯穿三个集合固定的，那么对于训练集，验证集分别增加的55个节点，以及测试集中最大可增加节点数110，总使用到的节点数为495+55+55+110=715<716,没有超过总节点数。

[Additional Notes]
I would like to add, take the SD dataset as an example, among the 550 nodes, only 495 nodes are fixed throughout the three sets, so for the training set, the validation set were increased by 55 nodes, as well as the maximum number of nodes that can be increased by 110 in the test set, the total number of nodes used is 495+55+55+110=715<716, not more than the total number of nodes.
